### PR TITLE
add merge range in get_ranges

### PR DESCRIPTION
**Pull request by Ted-Jiang**

### CODE REVIEW

File: hdfs/src/object_store/hdfs.rs [hunk 1]
```diff
@@ -440,6 +429,48 @@ fn convert_walkdir_result(
...
+) -> Result<Vec<Bytes>>
+where
+    F: FnMut(std::ops::Range<usize>) -> Fut,
+    Fut: std::future::Future<Output = Result<Bytes>>,
+{
+    let mut ret = Vec::with_capacity(ranges.len());
+    let mut start_idx = 0;
+    let mut end_idx = 1;
+
+    while start_idx != ranges.len() {
```
Comment (yahoNanJing) [inline hdfs/src/object_store/hdfs.rs]: We may need to sort the ranges before checking.

Comment (Ted-Jiang) [reply hdfs/src/object_store/hdfs.rs]: 👍 fix in [add sort](https://github.com/datafusion-contrib/datafusion-objectstore-hdfs/pull/10/commits/8fd32b61e9e2cfa63ca81a4482fab95ea66b9895)

Comment (Ted-Jiang) [reply hdfs/src/object_store/hdfs.rs]: @yahoNanJing

**Review by yahoNanJing — APPROVED** [review]
LGTM

**Pull request merged**